src/test/kotlin/no/nav/faktureringskomponenten/service/FakturaGeneratorTest.kt

```diff
@@ -265,12 +270,344 @@ class FakturaGeneratorTest {
+                    LocalDate.of(2024, 1, 1),
+                    LocalDate.of(2024, 3, 31),
+                    "Første periode"
+                ),
+                FakturaseriePeriode(
+                    BigDecimal(2000),
+                    LocalDate.of(2024, 3, 15), // Merk: Overlapper med første periode
+                    LocalDate.of(2024, 6, 30),
+                    "Andre periode"
+                )
+            ),
+            FakturaserieIntervall.KVARTAL
+        )
+
+        // Verifiser at vi får korrekt antall fakturalinjer og beløp
+        faktura.flatMap { it.fakturaLinje }.size.shouldBe(3) // Q1, overlapp periode, Q2
+    }
+
+    @Test
+    fun `periodisering som starter midt i en måned`() {
+        val dagensDato = LocalDate.of(2023, 12, 1)
+        mockkStatic(LocalDate::class)
+        every { LocalDate.now() } returns dagensDato
+
+        val faktura = generator.lagFakturaerFor(
+            periodisering = FakturaserieGenerator.genererPeriodisering(
+                LocalDate.of(2024, 1, 15), // Merk: Starter midt i måneden
+                LocalDate.of(2024, 6, 30),
+                FakturaserieIntervall.KVARTAL
+            ),
+            fakturaseriePerioder = listOf(
+                FakturaseriePeriode(
+                    BigDecimal(1000),
+                    LocalDate.of(2024, 1, 1),
+                    LocalDate.of(2024, 6, 30),
+                    "Test periode"
+                )
+            ),
+            FakturaserieIntervall.KVARTAL
+        )
+
+        // Verifiser at første fakturalinjes startdato er 15. januar
+        faktura.first().fakturaLinje.first().periodeFra.shouldBe(LocalDate.of(2024, 1, 15))
+    }
+
+    @Test
+    fun `fakturaseriePeriode som delvis overlapper med periodisering`() {
+        val dagensDato = LocalDate.of(2023, 12, 1)
+        mockkStatic(LocalDate::class)
+        every { LocalDate.now() } returns dagensDato
+
+        val faktura = generator.lagFakturaerFor(
+            periodisering = FakturaserieGenerator.genererPeriodisering(
+                LocalDate.of(2024, 1, 1),
+                LocalDate.of(2024, 3, 31),
+                FakturaserieIntervall.KVARTAL
+            ),
+            fakturaseriePerioder = listOf(
+                FakturaseriePeriode(
+                    BigDecimal(1000),
+                    LocalDate.of(2023, 12, 15), // Starter før periodisering
+                    LocalDate.of(2024, 4, 15),  // Slutter etter periodisering
+                    "Test periode"
+                )
+            ),
+            FakturaserieIntervall.KVARTAL
+        )
+
+        faktura.flatMap { it.fakturaLinje }.run {
+            first().periodeFra.shouldBe(LocalDate.of(2024, 1, 1)) // Skal starte med periodiseringen
+            last().periodeTil.shouldBe(LocalDate.of(2024, 3, 31)) // Skal slutte med periodiseringen
+        }
+    }
+
+    @Test
+    fun `dagensDato er nøyaktig på grensen mellom historisk og fremtidig`() {
```

Comment:
Ikke veldig tydelig hvorfor denne nesten er nyttig/nødvendig.

Reply:
Forbedret testnavn. Det er for å forsikre at det blir produsert to fakturaer, ikke én.